### PR TITLE
Add "show global status" support

### PR DESCRIPTION
**Pull request by sahutd**
Signed-off-by: Saimadhav Heblikar <saimadhavheblikar@gmail.com>

## Description

Sends show global status to any shard/any tablet in default keyspace. If no default keyspace, send to any shard/tablet in any keyspace. if no default keyspace, fallback to existing behavior.

## Related Issue(s)

https://github.com/vitessio/vitess/issues/6995

## Checklist
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtgate/planbuilder/show.go [hunk 1]
```diff
@@ -275,6 +275,23 @@ func buildPlanWithDB(show *sqlparser.ShowBasic, vschema ContextVSchema) (engine.
 
 }
 
+func buildShowStatusGlobalPlan(show *sqlparser.ShowBasic, vschema ContextVSchema) (engine.Primitive, error) {
```
Comment (harshit-gangal) [inline go/vt/vtgate/planbuilder/show.go:278]: nit: can be renamed to `buildShowStatusPlan` as it is building plan for global and session status.

File: go/vt/vtgate/planbuilder/show.go [hunk 2]
```diff
@@ -275,6 +275,23 @@ func buildPlanWithDB(show *sqlparser.ShowBasic, vschema ContextVSchema) (engine.
...
+	var ks *vindexes.Keyspace
+	var err error
+
+	if ks, err = vschema.AnyKeyspace(); err != nil {
+		// not returning err here for backward compatibility
+		return engine.NewRowsPrimitive(make([][]sqltypes.Value, 0, 2), buildVarCharFields("Variable_name", "Value")), nil
```
Comment (harshit-gangal) [inline go/vt/vtgate/planbuilder/show.go:284]: It is safe to return the error.

**Review by harshit-gangal — COMMENTED** [review]
looks good overall.
some comments. 
Also add tests in planbuilder package in show_cases and show_cases_no_default_keyspace

**Comment by sahutd** [comment]
@harshit-gangal addressed the comments. please take a look

File: go/vt/vtgate/planbuilder/show.go [hunk 3]
```diff
@@ -275,6 +275,22 @@ func buildPlanWithDB(show *sqlparser.ShowBasic, vschema ContextVSchema) (engine.
...
+	}
+
+	return &engine.Send{
+		Keyspace:          ks,
+		TargetDestination: key.DestinationAnyShard{},
+		Query:             sqlparser.String(show),
+		SingleShardOnly:   true,
+	}, nil
+}
+
```
Comment (harshit-gangal) [inline go/vt/vtgate/planbuilder/show.go:293]: this looks equivalent to `buildSendAnywherePlan`, it can be changed to use that function.

**Comment by harshit-gangal** [comment]
You would need to fix the DCO as well.
https://github.com/vitessio/vitess/pull/8344/checks?check_run_id=3096567516

File: go/vt/vtgate/planbuilder/show.go (hunk 3 above)
Comment (sahutd) [reply go/vt/vtgate/planbuilder/show.go]: yup. its exactly the same :)

**Comment by sahutd** [comment]
> You would need to fix the DCO as well.
> https://github.com/vitessio/vitess/pull/8344/checks?check_run_id=3096567516

@harshit-gangal addressed. thank you for your patience!

**Review by harshit-gangal — APPROVED** [review]

**Comment by harshit-gangal** [comment]
The test failure is related to the changes
```
--- FAIL: TestExecutorShow (0.00s)
```

**Comment by sahutd** [comment]
> The test failure is related to the changes
> 
> ```
> --- FAIL: TestExecutorShow (0.00s)
> ```

@harshit-gangal done. can you enable the tests to run please. afaik, all the failed actions had failed due to this 1 test only.

**Pull request merged**